src/index.js

```diff
@@ -193,6 +210,54 @@ class DropdownTreeSelect extends Component {
     this.keepDropdownActive = false
   }
 
+  onKeyboardKeyDown = e => {
+    const { readOnly, simpleSelect } = this.props
+    const { showDropdown, tags, searchModeOn, currentFocus } = this.state
+    const tm = this.treeManager
+    const tree = searchModeOn ? tm.matchTree : tm.tree
+
+    if (!showDropdown && (keyboardNavigation.isValidKey(e.key, false) || /^\w$/i.test(e.key))) {
+      // Triggers open of dropdown and retriggers event
+      e.persist()
+      this.handleClick(null, () => this.onKeyboardKeyDown(e))
+      if (/\w/i.test(e.key)) return
+    } else if (showDropdown && keyboardNavigation.isValidKey(e.key, true)) {
+      const newFocus = tm.handleNavigationKey(
+        currentFocus,
+        tree,
+        e.key,
+        readOnly,
+        !searchModeOn,
+        this.onCheckboxChange,
+        this.onNodeToggle
+      )
+      if (newFocus !== currentFocus) {
+        this.setState({ currentFocus: newFocus })
+      }
+    } else if (showDropdown && ['Escape', 'Tab'].indexOf(e.key) > -1) {
+      if (simpleSelect && tree.has(currentFocus)) {
+        this.onCheckboxChange(currentFocus, true)
+      } else {
+        // Triggers close
+        this.keepDropdownActive = false
+        this.handleClick()
+      }
+      return
+    } else if (
+      e.key === 'Backspace' &&
+      tags &&
```

Comment:
Is it possible to get rid of these (seemingly) excessive branches? 

Something like:
```
      e.key === 'Backspace' &&
      tags.length &&
      this.searchInput.value.length === 0
```

`tags` can be eliminated if we set it to the empty array in the constructor - not sure if that would have other ramifications and by the time this handler executes, `this.searchInput` should be valid.

Reply:
getTags seems safe to always return an array so should be fine.